pkg/sdkstats/sdkstats.go

```diff
@@ -2,161 +2,120 @@ package sdkstats
-	FullName     string
+	URL          string
+	Name         string
+	Owner        string
 	ProjectType  string
 	Stars        int
-	SDKVersion   int
+	SDKVersion   string
 	CreatedAt    string
 	PushedAt     string
 	TotalCommits int
 }
 
-//GetStats returns []RepoMetaData populated with Code results from Github Search Code API, for a given search string
-func GetStats(client *github.Client, q string) ([]RepoMetadata, error) {
-	var repoList []RepoMetadata
-	var projectType string
-	var sdkVersion int
-
-	//Search function return CodeResults for the q queery, and also Basic repository data.
-	searchOp, err := Search(client, q)
-
-	if err != nil {
-		return nil, err
-	}
-
-	if strings.Contains(q, "helm") {
-		projectType = "helm"
-	}
-	if strings.Contains(q, "ansible") {
-		projectType = "ansible"
-	}
-	if strings.Contains(q, "go.mod") {
-		projectType = "go.mod"
-	}
-	if strings.Contains(q, "Gopkg.toml") {
-		projectType = "Gopkg.toml"
-	}
-	for j := 0; j < len(searchOp); j++ {
-		for i := 0; i < len(searchOp[j].CodeResults); i++ {
-			owner := searchOp[j].CodeResults[i].GetRepository().GetOwner().GetLogin()
-			name := searchOp[j].CodeResults[i].GetRepository().GetName()
-			fullName := searchOp[j].CodeResults[i].GetRepository().GetFullName()
-
-			//getVersion returns SDK version by doing Fragment search
-			if projectType == "Gopkg.toml" {
-				sdkVersion = 0
-			} else {
-				sdkVersion = getversion(searchOp[j].CodeResults[i], projectType)
-			}
+//RepoMetadataQuery struct to store Search queries, and Project Type
+type RepoMetadataQuery struct {
+	ProjectType   string
+	Queries       []string
+	VersionParser VersionParser
+}
 
-			//GetRepoDetails returns Repository specififc details
-			repoMap := GetRepoDetails(client, owner, name)
+//VersionParser Interface implements ParseVersion, to Parse Version number from a given Text Match result.
+type VersionParser interface {
+	ParseVersion(codeResults github.CodeResult, projectType string) (string, error)
+}
 
-			TotalCommits, err := strconv.Atoi(repoMap["TotalCommits"])
-			Stars, err := strconv.Atoi(repoMap["Stars"])
-			if err != nil {
-				return nil, err
-			}
+//GetStats returns []RepoMetaData populated with Code results from Github Search Code API, for a given search string
+func GetStats(client *github.Client, rq RepoMetadataQuery) ([]RepoMetadata, error) {
+	var repoList []RepoMetadata
+	var sdkVersion string
+	var skipVersion int
 
-			repoDetails := &RepoMetadata{
-				FullName:     fullName,
-				ProjectType:  projectType,
-				SDKVersion:   sdkVersion,
-				Stars:        Stars,
-				CreatedAt:    repoMap["CreatedAt"],
-				PushedAt:     repoMap["PushedAt"],
-				TotalCommits: TotalCommits,
+	for _, q := range rq.Queries {
+		searchOp, err := Search(client, q)
+		if err != nil {
+			return nil, err
+		}
+		if strings.Contains(q, "Gopkg.toml") {
+			skipVersion = 1
+		}
+		for j := 0; j < len(searchOp); j++ {
+			for i := 0; i < len(searchOp[j].CodeResults); i++ {
```

Comment:
It doesn't look like you need the indices here. Go has a shortcut:

```go
for _, searchResult := range searchOp {
    for _, codeResult := range searchResult.CodeResults {
        version, err := rq.VersionParser.ParseVersion(codeResult)
        if err != nil {
            return nil, err
        }
        ....
    }
}

Reply:
Yes, I  had range for loop earlier, However, for some weirs reason, it omitted last page of results. I am not sure why though. Thats why i switched to indices. To be honest, I did not put after thought into it to investigate.

Reply:
It's probably worth it to retry with `for .. range` since that's the convention. There should be no difference between the loops unless the loop contains code that changes the contents of the slice, which doesn't seem to be the case here.

Reply:
Done